src/networking.c

```diff
@@ -3752,3 +3863,52 @@ int handleClientsWithPendingReadsUsingThreads(void) {
 
     return processed;
 }
+
+/* Returns true if client memory limit was reached */
+int clientEvictionCheckLimit() {
+    size_t maxmemory_clients_actual = SIZE_MAX;
+
+    /* Handle percentage of maxmemory*/
+    if (server.maxmemory_clients < 0 && server.maxmemory > 0)
+        maxmemory_clients_actual = (size_t)((double)server.maxmemory * -(double)server.maxmemory_clients / 100);
+    else if (server.maxmemory_clients > 0)
+        maxmemory_clients_actual = server.maxmemory_clients;
+    else
+        return 0;
+
+    /* Don't allow a too small maxmemory-clients to avoid cases where we can't communicate
+     * at all with the server because of bad configuration */
+    if (maxmemory_clients_actual < 1024*128)
+        maxmemory_clients_actual = 1024*128;
```

Comment:
I think calc  `maxmemory_clients_actual` can be a separate method, and if it returns 0, we can eliminate the need for the while loop in `evictClients`.

Reply:
:+1: